src/Model/KeyDisplayer.re

```diff
@@ -7,96 +7,97 @@
 // We group key presses in time together,
 // so they show in the same line
 type groupedPresses = {
-  exclusive: bool,
+  isExclusive: bool,
   time: float,
   keys: list(string),
 };
 
 type t = {
   // Enabled is whether or not the key displayer feature is turned on
-  enabled: bool,
+  isEnabled: bool,
   // Active is whether or not there is a visible key press.
-  active: bool,
+  isActive: bool,
   presses: list(groupedPresses),
 };
 
-let empty: t = {active: false, presses: [], enabled: false};
+let initial: t = {isActive: false, presses: [], isEnabled: false};
 
 module Constants = {
   let timeToShow = 2.5;
   let timeToGroup = 0.3;
 };
 
-let setEnabled = (enabled, _v: t) => {
-  {...empty, enabled};
+let enable = _model => {
+  {...initial, isEnabled: true};
 };
 
-let update = (time, v: t) => {
-  let f = k => time -. k.time < Constants.timeToShow;
-  let presses = List.filter(f, v.presses);
-  let active = List.length(presses) > 0;
-  {...v, active, presses};
+let collectGarbage = (time, model) => {
+  let presses =
+    List.filter(
+      press => time -. press.time < Constants.timeToShow,
+      model.presses,
+    );
+  let isActive = List.length(presses) > 0;
+  {...model, isActive, presses};
 };
 
-let add = (time, key, v: t) => {
-  let exclusive = String.length(key) > 1 || key == " ";
+let add = (time, key, model) => {
+  let isExclusive = String.length(key) > 1 || key == " ";
   let presses =
-    switch (v.presses) {
-    | [] => [{time, exclusive, keys: [key]}]
+    switch (model.presses) {
+    | [] => [{time, isExclusive, keys: [key]}]
     | [hd, ...tail] =>
       if (time
           -. hd.time <= Constants.timeToGroup
-          && !hd.exclusive
+          && !hd.isExclusive
           && String.length(key) == 1) {
         [
           // The key presses was within the group time,
           // so we'll just add it to an existing group
-          {time, exclusive, keys: [key, ...hd.keys]},
+          {time, isExclusive, keys: [key, ...hd.keys]},
           ...tail,
         ];
       } else {
-        let exclusive = String.length(key) > 1;
+        let isExclusive = String.length(key) > 1;
         [
           // The time was past the group time..
           // so we'll create a new group
-          {time, exclusive, keys: [key]},
+          {time, isExclusive, keys: [key]},
           hd,
           ...tail,
         ];
       }
     };
 
-  let ret = {...v, active: true, presses};
+  let ret = {...model, isActive: true, presses};
 
   // Also filter out old key presses, while we're here
-  update(time, ret);
+  collectGarbage(time, ret);
```

Comment:
I like this more explicit name better than `update` - for a second though I thought it was referring to the actual OCaml garbage collector though (ie, `GC.full_major`). Maybe `removeExpired` would be clearer?

Reply:
Yeah that does sound better. Good suggestion!